pandas/core/indexes/multi.py

```diff
@@ -2328,23 +2337,40 @@ def convert_indexer(start, stop, step, indexer=indexer, labels=labels):
             j = labels.searchsorted(loc, side='right')
             return slice(i, j)
 
-    def get_locs(self, tup):
+    def get_locs(self, seq):
         """
-        Given a tuple of slices/lists/labels/boolean indexer to a level-wise
-        spec produce an indexer to extract those locations
+        Get location for a given label/slice/list/mask or a sequence of such as
+        a integer, slice or boolean mask.
 
         Parameters
         ----------
-        key : tuple of (slices/list/labels)
+        seq : label/slice/list/mask or a sequence of such.
 
         Returns
         -------
-        locs : integer list of locations or boolean indexer suitable
+        locs : integer, slice or boolean mask suitable
```

Comment:
I don't think this is correct: you changed 'integer list of locations' to 'integer', but from the examples it seems to return an array of integers, so the 'list' was more correct

Reply:
Well, tup indicates that this method takes a tuple, but it also takes other sequences like lists. In an example I tried it actually choked on tuple, because it thought the tuple was the label of a level. My experience from the last days it therefore is that it's probaby best to use lists rather than tuples (but tuple may work, depending on context).

On the other hand get_loc (without s) demands a tuple and chokes on lists, so it's not so easy to remember what is accepted where.

I can say "list or such", to discourage using tuples?

Reply:
Note that my comment was about the *return value*, not the input (and there indeed tuples vs lists sometimes are slightly different but not always, which can be confusing)

Reply:
Ok, changed

Reply:
Note, I actually couldn't manage to get it to return a boolean mask.

Could you confirm that? I haven't changed that part, but if it's not true, it'd be better it didn't say that...

Reply:
Yes, looking at the code, I actually think it can only return an integer array. That would mean that the explanation is just wrong.

But I am not too familiar with this. (@jreback ?)